src/agents/base/AgentExecuter.ts

```diff
@@ -5,22 +5,22 @@ export class AgentExecuter {
   objective: string;
   modelName: string;
   messageCallback: (message: Message) => void;
-  statusCallback: (status: AgentStatus) => void;
+  statusCallback: (status: AgentStatus, taskId?: number | string | undefined | null) => void;
   cancelCallback: () => void;
   verbose: boolean = false;
-  //
+
   taskIdCounter: number = 0;
   taskList: AgentTask[] = [];
-  isRunning: boolean = false;
-  //
+  isRunning!: boolean | string;
```

Comment:
I think we don't especially, but how you would like to manage... I mean it can be only boolean, then a pause or reverse method by an other way referenced and connected to this variable. I was just thinking about that way, if the isRunning state managed like a "multioption". Maybe I was wrong and I overthinked already a little.

Reply:
Okay, I think you are looking ahead so I don't think it is a mistake. I just don't want to overdo it, so let's change it if it becomes necessary.